substrate/frame/assets/src/functions.rs

```diff
@@ -575,7 +607,8 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		// Execute hook outside of `mutate`.
 		if let Some(Remove) = target_died {
-			T::Freezer::died(id, target);
+			T::Freezer::died(id.clone(), target);
+			T::Holder::died(id, target);
```

Comment:
This method is called from `burn`. I don't where it is check that the account is allowed to die. Are we ok that admin can kill the account even when there some on hold?

Reply:
Not really.

This is the exact reason why we had such a nightmare working on the ensures before an account could die.

This is a hook, not intended to _burn up_ resources, but to make proper cleanup steps on external systems (if any needed) whenever an account is marked as dead (hence, informing the account already `died`).

That's also why its `Freezer` counterpart is also called that way.

Reply:
should be ensure no freezes and holds for `decrease_balance` as well

Reply:
Done on [`a37b00b`](https://github.com/paritytech/polkadot-sdk/pull/4530/commits/a37b00b323bedfa16617e311d26989e72075186b).